src/server/replica.cc

```diff
@@ -1023,7 +1030,21 @@ Replica::Info Replica::GetInfo() const {
     res.full_sync_done = (state_mask_.load() & R_SYNC_OK);
     res.master_last_io_sec = (ProactorBase::GetMonotonicTimeNs() - last_io_time) / 1000000000UL;
     return res;
-  });
+  };
+
+  if (ABSL_PREDICT_TRUE(Sock()))
+    return Proactor()->AwaitBrief(f);
+  else
+    return ProactorBase::me()->AwaitBrief(f);
+  /**
+   * when this branch happens: there is a very short grace period
+   * where Sock() is not initialized, yet the server can
+   * receive ROLE/INFO commands. That period happens when launching
+   * an instance with '--replicaof' and then \b immediately
+   * sending a command.
+   *
+   * In that instance, we have to run f() on the current fiber.
```

Comment:
But what's the downside of running in current fiber? Why not always do it as such? And why not run it just via `f()` instead of awaiting for our own thread?

Reply:
that I am not entirely sure about. Presumably this was done in order to not block the current fiber, the same fiber that runs `MainReplicationFb`.

About why I run `f()` using `AwaitBrief`, I was not sure that was the same outcome. I'll change it, thanks!